### PR TITLE
fix: add propose transfer functions

### PR DESCRIPTION
**Pull request by 0xGrapefruit**

### CODE REVIEW

File: src/Pod.ts [hunk 1]
```diff
@@ -206,6 +211,49 @@ export default class Pod {
...
 
+  /**
+   * Transfers a membership from the signer's account to the memberToTransferTo.
+   * @param addressToTransferTo
+   * @param signer
+   * @returns
+   */
+  transferMembership = async (addressToTransferTo: string, signer: ethers.Signer) => {
+    const checkedAddress = checkAddress(addressToTransferTo);
+    const signerAddress = await signer.getAddress();
```
[the remark this answers is not in the document]
Comment (jtlacci) [reply src/Pod.ts:222]: and a `!isMember` for the to address ?

File: src/Pod.ts [hunk 2]
```diff
@@ -384,4 +432,104 @@ export default class Pod {
...
+    // Sub pod must be the admin or a subpod of this pod.
+    if (!(await this.isMember(subPod.safe))) {
+      throw new Error(`Pod ${subPod.ensName} must be a subpod of this pod to make proposals`);
+    }
+
+    const signerAddress = await signer.getAddress();
+    if (!(await subPod.isMember(signerAddress)))
+      throw new Error(`Signer ${signerAddress} was not a member of sub pod ${subPod.ensName}`);
+
+    // Tells MemberToken to mint a new token for this pod to newMember.
```
Comment (jtlacci) [inline src/Pod.ts:462]:
```js
// Tells MemberToken to transfer token for this pod from subpod.safe to checkedAddress.
```

File: src/Pod.ts [hunk 3]
```diff
@@ -384,4 +432,104 @@ export default class Pod {
...
+      throw new Error(`Pod ${adminPod.ensName} must be the admin of this pod`);
+    }
+
+    const signerAddress = await signer.getAddress();
+    if (!(await adminPod.isMember(signerAddress)))
+      throw new Error(`Signer ${signerAddress} was not a member of admin pod ${adminPod.ensName}`);
+
+    const { abi: controllerAbi } = getControllerByAddress(this.controller, config.network);
+    const data = new ethers.utils.Interface(controllerAbi).encodeFunctionData('updatePodAdmin', [
+      1,
```
Comment (jtlacci) [inline src/Pod.ts:515]: should be `this.id`

**Review by jtlacci — CHANGES_REQUESTED** [review]
feel like maybe we should figure out a way to do some more stateful integration tests

File: src/Pod.ts [hunk 4]
```diff
@@ -384,4 +432,104 @@ export default class Pod {
...
+      throw new Error(`Pod ${subPod.ensName} must be a subpod of this pod to make proposals`);
+    }
+
+    const signerAddress = await signer.getAddress();
+    if (!(await subPod.isMember(signerAddress)))
+      throw new Error(`Signer ${signerAddress} was not a member of sub pod ${subPod.ensName}`);
+
+    // Tells MemberToken to mint a new token for this pod to newMember.
+    const data = encodeFunctionData('MemberToken', 'safeTransferFrom', [
+      signerAddress,
```
[the remark this answers is not in the document]
Comment (0xGrapefruit) [reply src/Pod.ts:464]: Good catch

**Review by jtlacci — APPROVED** [review]

**Pull request merged**